server/src/modules/runs/runs.service.ts

```diff
@@ -64,4 +65,18 @@ export class RunsService {
 
         return new PaginatedResponseDto(RunDto, dbResponse);
     }
+
+    async deleteStoredMapRuns(mapID: number): Promise<void> {
+        const where: Prisma.RunWhereInput = {
+            mapID
+        };
+        const dbResponse = await this.runRepo.getAllRuns(where);
+        const runs = dbResponse.flatMap<Run>((resp) => resp[0]);
```

Comment:
I'm really confused by what you're doing here. `getAllRuns` returns a `[Run[], number]` tuple, so mapping the elements of the tuple to some subelement is going to break for the second value (the count). Out of curiosity I checked, and in JS trying to access the index of a number just returns `undefined`, regardless it's certainly not something we should be doing. It will definitely fail tests once they're written.

You just want the arrays of runs, so you only need `dbResponse[0]`. I would probably just do `const [runs] = await this.runRepo.getAllRuns(where);` since `(await this.runRepo.getAllRuns(where))[0];` looks rather unpleasant.